archinstall/lib/installer.py

```diff
@@ -844,63 +847,77 @@ def add_systemd_bootloader(self, boot_partition :Partition, root_partition :Part
 		if not os.path.exists(f'{self.target}/boot/loader/entries'):
 			os.makedirs(f'{self.target}/boot/loader/entries')
 
+		comments = [
+				'# Created by: archinstall\n',
+				'# Created on: {}\n'.format(self.init_time)
+		]
+
+		contents = [
+				'title   Arch Linux ({kernel})\n',
+				'linux   /vmlinuz-{kernel}\n',
+				'initrd  /initramfs-{kernel}.img\n'
+		]
```

Comment:
Would it be cleaner to do:
```python
contents = [
		'title   Arch Linux ({kernel})',
		'linux   /vmlinuz-{kernel}',
		'initrd  /amd-ucode.img' if vendor == 'AuthenticAMD' else '',
		'initrd  /intel-ucode.img' if vendor == 'GenuineIntel' else '',
		'initrd  /initramfs-{kernel}.img'
]
```
And also do `'\n'.join(contents).format(kernel=kernel)` instead to remove the multiple `\n`?

Reply:
There are issues with that solution. Here is a demonstration:

```
>>> val = 'B'
>>> l = [
...     'a',
...     'b' if val == 'B' else '',
...     'c' if val == 'C' else '',
...     'd'
... ]
>>> print(l)
['a', 'b', '', 'd']
>>> print('\n'.join(l))
a
b

d
>>> val = 'X'
>>> l = [
...     'a',
...     'b' if val == 'B' else '',
...     'c' if val == 'C' else '',
...     'd'
... ]
>>> print(l)
['a', '', '', 'd']
>>> print('\n'.join(l))
a


d
```

The portion of code that currently takes care of enabling microcode also serves to call `log()` when a match for microcode is not found.

The intention behind using `\n` in the list items is to be explicit that it is a list of lines.

Reply:
That's true. I was mainly trying to conserve code complexity and amount of code since this PR would clean up a lot of the legacy nastiness that we have.

The `log()` would be hard to get around, but the empty newlines could be circumvented with `print('\n'.join(x for x in l if x))`. But that would add complexity I suppose.

It's good enough for me, just bouncing ideas :) Great work!